akka-stream/src/main/scala/akka/stream/impl/ReactiveStreamsCompliance.scala

```diff
@@ -125,13 +126,14 @@ import org.reactivestreams.{ Subscriber, Subscription }
     }
   }
 
-  final def tryCancel(subscription: Subscription): Unit = {
+  final def tryCancel(subscription: Subscription, cause: Throwable): Unit = {
     if (subscription eq null)
       throw new IllegalStateException("Subscription must be not null on cancel() call, rule 1.3")
-    try subscription.cancel()
-    catch {
-      case NonFatal(t) =>
-        throw new SignalThrewException("It is illegal to throw exceptions from cancel(), rule 3.15", t)
+    try subscription match {
+      case s: SubscriptionWithCancelException ⇒ s.cancel(cause)
+      case s ⇒ s.cancel()
+    } catch {
+      case NonFatal(t) ⇒ throw new SignalThrewException("It is illegal to throw exceptions from cancel(), rule 3.15", t)
```

Comment:
Surprised `⇒` made it through

Reply:
The original change was quite old and before the reformatting, seems scalafmt doesn't convert it automatically.

Reply:
Replaced.